guide.py

```diff
@@ -0,0 +1,434 @@
+
+
+# Forms to be used for each stage of the process.
+STAGE_FORMS = {
+    models.INTENT_NONE: guideforms.Incubate,
+    models.INTENT_IMPLEMENT: guideforms.Prototype,
+    models.INTENT_EXPERIMENT: guideforms.DevTrial,
+    models.INTENT_EXTEND_TRIAL: guideforms.OriginTrial,
+}
+
+class FeatureNew(common.ContentHandler):
+
+  def get(self, path):
+    user = users.get_current_user()
+    if user is None:
+      return self.redirect(users.create_login_url(self.request.uri))
+
+    # TODO(ericbidelman): This creates a additional call to
+    # _is_user_whitelisted() (also called in common.py), resulting in another
+    # db query.
+    if not self._is_user_whitelisted(user):
+      #TODO(ericbidelman): Use render(status=401) instead.
+      #self.render(data={}, template_path=os.path.join(path + '.html'), status=401)
+      common.handle_401(self.request, self.response, Exception)
+      return
+
+    template_data = {
+        'overview_form': guideforms.OverviewForm(),
+        }
+
+    self._add_common_template_values(template_data)
+
+    self.render(data=template_data, template_path=os.path.join(path + '.html'))
+
+  def post(self, path):
+    user = users.get_current_user()
+    if user is None or (user and not self._is_user_whitelisted(user)):
+      common.handle_401(self.request, self.response, Exception)
+      return
+
+    feature = models.Feature(
+        category=int(self.request.get('category')),
+        name=self.request.get('name'),
+        intent_stage=models.INTENT_NONE,
+        summary=self.request.get('summary'),
+        impl_status_chrome=models.NO_ACTIVE_DEV,
+        visibility=models.WARRANTS_ARTICLE,
+        standardization=models.EDITORS_DRAFT,
+        web_dev_views=models.DEV_NO_SIGNALS)
+    key = feature.put()
+
+    # TODO(ericbidelman): enumerate and remove only the relevant keys.
+    memcache.flush_all()
+
+    redirect_url = '/guide/edit/' + str(key.id())
+    return self.redirect(redirect_url)
+
+
+class ProcessOverview(common.ContentHandler):
+
+  def get(self, path, feature_id):
+    user = users.get_current_user()
+    if user is None:
+      # Redirect to public URL for unauthenticated users.
+      return self.redirect(self.DEFAULT_URL + '/' + feature_id)
+
+    # TODO(ericbidelman): This creates a additional call to
+    # _is_user_whitelisted() (also called in common.py), resulting in another
+    # db query.
+    if not self._is_user_whitelisted(user):
+      common.handle_401(self.request, self.response, Exception)
+      return
+
+    template_data = {
+        'overview_form': guideforms.OverviewForm(),
+        'process_json': json.dumps(
+            [stage._asdict() for stage in processes.BLINK_PROCESS]),
+        'progress_so_far': [],
+        }
+
+    f = models.Feature.get_by_id(long(feature_id))
+    if f is None:
+      return self.redirect(self.ADD_NEW_URL)
+
+    progress_so_far = []  # An unordered list of progress item strings.
+    progress_so_far = ['Explainer', 'API design']
```

Comment:
Can remove the line instantiating the empty list (it's immediately replaced w/ another object).

Reply:
That second line is a placeholder for where I will need to add some logic to detect which progress items have been done so far.  I added a TODO comment to make that explicit.  I'll resolve that TODO in an upcoming CL.